### PR TITLE
ports/rp2: Add board overclock config.

### PR DESCRIPTION
**Pull request by Gadgetoid**
This is a first shot at implementing #8208

Some points for discussion:

* Naming things, ugh, what should these variables be called?
* The 10ms delay for vreg stabilization is... entirely without experimentation. It's what we use in PicoSystem but that number may have to be higher/lower for other frequencies. Looking at people pushing the limits of clock speed and there are delays anywhere from 1 to 1000ms. There's no SDK function or register we can poll to wait for a stable vreg. (@kilograham do you have any insight into a sensible approach here?)
* Should the delay even happen if the vreg is set to default? It could be optimized out with an ifdef.

It's worth noting that since `machine.freq(250_000_000)` does not apply an overvolt, it's suboptimal for actually *overclocking* the RP2040 since some chips (we found this out when testing PicoSystem) do not run stably without at least a modest overvolt. A default overclock for boards that benefit from it would "guarantee" (insomuch as we can do so) that a board will run at the given frequency.

The change intends to clean up this hack - https://github.com/pimoroni/micropython/commit/10c57803c032754e4323977cc815e24e092a4a5d

This will permit us to ship some more overclocked MicroPython firmware variants for boards where it makes sense- mostly stuff that does a lot of pixel-pushing like Galactic Unicorn, PicoSystem and Plasma 2040. The end-user can always use `machine.freq()` to *downclock* if they prefer power savings over raw performance.

### CODE REVIEW

**Comment by Gadgetoid** [comment]
This should probably use the Pico SDK defines for clock frequency, if it does anything at all.

Additionally only values computed via vcocalc.py can/will work: https://github.com/raspberrypi/pico-sdk/blob/2.1.1/src/rp2_common/hardware_clocks/scripts/vcocalc.py

vcocalc.py produces the following values, intended to be used with CMake's `target_compile_definitions`:

* `PLL_SYS_REFDIV`
* `PLL_SYS_VCO_FREQ_HZ`
* `PLL_SYS_POSTDIV1`
* `PLL_SYS_POSTDIV2`
* `SYS_CLK_HZ`

Using the default refdiv, some clock frequencies are unachievable (this manifests at runtime in `machine.freq()` as a `ValueError: cannot change frequency`) and compiling with a different `PLL_SYS_REFDIV` would change the clock frequencies accepted by `machine.freq()`.

AIUI it's possible to achieve a (non-overvolt) default overclock by just specifying these values at compile time.

I also believe that these compile-time values can potentially be changed at runtime, though expanding `machine.freq()` to calculate them would cost precious bytes.

Anyway, TLDR:

Overvolt notwithstanding, it's possible that the following in `mpconfigboard.cmake` already just works:

```cmake
add_compile_definitions(
    PLL_SYS_REFDIV=1
    PLL_SYS_VCO_FREQ_HZ=1200000000
    PLL_SYS_POSTDIV1=6
    PLL_SYS_POSTDIV2=1
    SYS_CLK_HZ=200000000
)
```

Or for RP2040 at the manufacturer approved 200MHz, just:

```cmake
add_compile_definitions(
    PICO_USE_FASTEST_SUPPORTED_CLOCK=1
)
```

Making this PR largely redundant, save for some means of optionally overriding the voltage.

**Comment by Gadgetoid** [comment]
I think this is fully redundant; adding the above settings into our SDK level board headers has proven to be a sensible approach.
